src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveTestsWithVariousTypes.cs

```diff
@@ -0,0 +1,220 @@
+            }
+        }
+
+        private static void AssertMembers(Type[] types, Type[] typesFromDisk)
+        {
+            Assert.Equal(types.Length, typesFromDisk.Length);
+
+            for (int i = 0; i < types.Length; i++)
+            {
+                Type sourceType = types[i];
+                Type typeFromDisk = typesFromDisk[i];
+
+                AssertTypeProperties(sourceType, typeFromDisk);
+                AssertMethods(sourceType.GetMethods(), typeFromDisk.GetMethods());
+                AssertFields(sourceType.GetFields(), typeFromDisk.GetFields());
+            }
+        }
+
+        private static void AssertFields(FieldInfo[] declaredFields, FieldInfo[] fieldsFromDisk)
+        {
+            Assert.Equal(declaredFields.Length, fieldsFromDisk.Length);
+
+            for (int j = 0; j < declaredFields.Length; j++)
+            {
+                FieldInfo sourceField = declaredFields[j];
+                FieldInfo fieldFromDisk = fieldsFromDisk[j];
+
+                Assert.Equal(sourceField.Name, fieldFromDisk.Name);
+                Assert.Equal(sourceField.Attributes, fieldFromDisk.Attributes);
+                Assert.Equal(sourceField.FieldType.FullName, fieldFromDisk.FieldType.FullName);
+            }
+        }
+
+        private static void AssertMethods(MethodInfo[] sourceMethods, MethodInfo[] methodsFromDisk)
+        {
+            Assert.Equal(sourceMethods.Length, methodsFromDisk.Length);
+
+            for (int j = 0; j < sourceMethods.Length; j++)
+            {
+                MethodInfo sourceMethod = sourceMethods[j];
+                MethodInfo methodFromDisk = methodsFromDisk[j];
+
+                Assert.Equal(sourceMethod.Name, methodFromDisk.Name);
+                Assert.Equal(sourceMethod.Attributes, methodFromDisk.Attributes);
+                Assert.Equal(sourceMethod.ReturnType.FullName, methodFromDisk.ReturnType.FullName);
+            }
+        }
+
+        private static void AssertTypeProperties(Type sourceType, Type typeFromDisk)
+        {
+            Assert.Equal(sourceType.Name, typeFromDisk.Name);
+            Assert.Equal(sourceType.Namespace, typeFromDisk.Namespace);
+            Assert.Equal(sourceType.Attributes, typeFromDisk.Attributes);
+            Assert.Equal(sourceType.IsInterface, typeFromDisk.IsInterface);
+            Assert.Equal(sourceType.IsValueType, typeFromDisk.IsValueType);
+        }
+
+        [Theory]
+        [MemberData(nameof(VariousInterfacesStructsTestData))]
+        public void WriteAndReadVariousTypesToFromStream(Type[] types)
+        {
+            using var stream = new MemoryStream();
+            AssemblyTools.WriteAssemblyToStream(s_assemblyName, types, stream);
+            Assembly assemblyFromDisk = AssemblyTools.LoadAssemblyFromStream(stream);
+            Module moduleFromDisk = assemblyFromDisk.Modules.First();
+
+            Assert.Equal(s_assemblyName.Name, assemblyFromDisk.GetName().Name);
```

Comment:
"FromDisk" suffix does not make sense here. We have a memory stream that is not saved to disk. It would change the suffixes throughout the file to "expected" and "actual" that are names typically used for xunit equals assertions.